### PR TITLE
add press summary and press summary approval

### PR DESCRIPTION
**Pull request by dreamfall**

### CODE REVIEW

File: app/controllers/concerns/press_summary_mixin.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        redirect_to [namespace_name, @form_answer]
+      end
+
+      format.js { render "admin/press_summaries/create" }
+    end
+  end
+
+  def press_summary_params
+    params.require(:press_summary).permit(:body)
+g  end
```
Comment (matthewford) [inline app/controllers/concerns/press_summary_mixin.rb]: `end` at 43, 3 is not aligned with `def` at 41, 2<br>Unnecessary spacing detected.<br>Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 2]
```diff
@@ -46,6 +46,7 @@ class FormAnswer < ActiveRecord::Base
     has_one :promotion_eligibility, class_name: 'Eligibility::Promotion', dependent: :destroy
     has_one :audit_certificate, dependent: :destroy
     has_one :feedback, dependent: :destroy
+    has_one :press_summary, dependent: :destroy
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/policies/press_summary_policy.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  def update?
+    return @can_update unless @can_update.nil?
+
+    @can_update = if assessor?
+      subject.lead?(form_answer) || (subject.assigned?(form_answer) && !record.approved?)
+    else
+      admin?
+    end
```
Comment (matthewford) [inline app/policies/press_summary_policy.rb]: `end` at 13, 4 is not aligned with `if` at 9, 18

File: app/policies/press_summary_policy.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  alias :create? :update?
+
+  def approve?
+    return @can_approve unless @can_approve.nil?
+
+    @can_approve = if assessor?
+      subject.lead?(form_answer) && !record.approved?
+    else
+      admin? && !record.approved?
+    end
```
Comment (matthewford) [inline app/policies/press_summary_policy.rb]: `end` at 24, 4 is not aligned with `if` at 20, 19

File: app/policies/press_summary_policy.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  def create?
+    admin? || (assessor? && subject.assigned?(form_answer))
+  end
+
+  def update?
+    return @can_update unless @can_update.nil?
+
+    @can_update = if assessor?
+      subject.lead?(form_answer) || (subject.assigned?(form_answer) && !record.approved?)
+    else
```
Comment (matthewford) [inline app/policies/press_summary_policy.rb]: Align `else` with `if`.

File: app/policies/press_summary_policy.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    end
+  end
+  alias :create? :update?
+
+  def approve?
+    return @can_approve unless @can_approve.nil?
+
+    @can_approve = if assessor?
+      subject.lead?(form_answer) && !record.approved?
+    else
```
Comment (matthewford) [inline app/policies/press_summary_policy.rb]: Align `else` with `if`.

File: app/policies/press_summary_policy.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
+class PressSummaryPolicy < ApplicationPolicy
+  def create?
+    admin? || (assessor? && subject.assigned?(form_answer))
+  end
+
+  def update?
+    return @can_update unless @can_update.nil?
+
+    @can_update = if assessor?
+      subject.lead?(form_answer) || (subject.assigned?(form_answer) && !record.approved?)
```
Comment (matthewford) [inline app/policies/press_summary_policy.rb]: Use 2 (not -12) spaces for indentation.

File: app/policies/press_summary_policy.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+      admin?
+    end
+  end
+  alias :create? :update?
+
+  def approve?
+    return @can_approve unless @can_approve.nil?
+
+    @can_approve = if assessor?
+      subject.lead?(form_answer) && !record.approved?
```
Comment (matthewford) [inline app/policies/press_summary_policy.rb]: Use 2 (not -13) spaces for indentation.

File: spec/factories/press_summaries.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+FactoryGirl.define do
+  factory :press_summary do
+    form_answer nil
+body "MyText"
+comment "MyText"
+approved false
+  end
+
```
Comment (matthewford) [inline spec/factories/press_summaries.rb]: Extra empty line detected at block body end.

File: spec/factories/press_summaries.rb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+FactoryGirl.define do
+  factory :press_summary do
+    form_answer nil
+body "MyText"
```
Comment (matthewford) [inline spec/factories/press_summaries.rb]: Inconsistent indentation detected.

File: spec/factories/press_summaries.rb [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+comment "MyText"
```
Comment (matthewford) [inline spec/factories/press_summaries.rb]: Inconsistent indentation detected.

File: spec/factories/press_summaries.rb [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+approved false
```
Comment (matthewford) [inline spec/factories/press_summaries.rb]: Inconsistent indentation detected.

File: app/controllers/concerns/press_summary_mixin.rb [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+    respond_to do |format|
+      format.html do
+        flash.notice = "Press Summary was successfully approved"
+        redirect_to [namespace_name, @form_answer]
+      end
+
+      format.js { render "admin/press_summaries/create" }
+    end
+  end
+
```
Comment (jaleszek) [inline app/controllers/concerns/press_summary_mixin.rb]: `private` missing ?

Comment (dreamfall) [inline app/controllers/concerns/press_summary_mixin.rb]: yeah

File: db/migrate/20150325160755_create_press_summaries.rb [hunk 14]
```diff
@@ -0,0 +1,13 @@
+class CreatePressSummaries < ActiveRecord::Migration
+  def change
+    create_table :press_summaries do |t|
+      t.references :form_answer, index: true
```
Comment (jaleszek) [inline db/migrate/20150325160755_create_press_summaries.rb]: does it can be null ?

Comment (dreamfall) [inline db/migrate/20150325160755_create_press_summaries.rb]: No, it's validates with rails.

Comment (jaleszek) [inline db/migrate/20150325160755_create_press_summaries.rb]: I see a lot benefits in validating it in db schema.

**Comment by jaleszek** [comment]
great :+1:

**Pull request merged**